bitcoin/src/blockdata/opcodes.rs

```diff
@@ -16,11 +16,11 @@
 use core::{fmt, convert::From};
 use bitcoin_internals::debug_from_display;
 
-// Note: I am deliberately not implementing PartialOrd or Ord on the
-//       opcode enum. If you want to check ranges of opcodes, etc.,
-//       write an #[inline] helper function which casts to u8s.
-
 /// A script Opcode.
+///
+/// We deliberately do not implement `PartialOrd` or `Ord` because there is no ordering of opcodes
+/// that makes conceptual sense. Note however that Bitcoin Core does use `<` and `>` on opcodes to
+/// effect consensus logic, something which hides implicit assumptions about the opcodes' byte encoding.
```

Comment:
In c56e54a1b4b9c22d9d4d8a73d7696b9c9544a3d4:

I'm happy to ACK this but TBH I don't think it belongs in a doccomment. Core's logic is hidden inside Core, so I think if you care about this you're probably reading the source code, not the API docs, and a regular comment will suffice.

Reply:
Demoted it back to a code comment. This PR wins this months prize for most iterations on a totally trivial PR :)

Reply:
I think the part about deliberately not implementing `Ord` is fine in doc comments so that the users will not ask us to implement it (or find the comment too late). For notes about Core a normal comment is fine. Notice that I wrote only the first part in my suggestion.

@tcharding perhaps you can save some time by waiting for resolution of our discussions before implementing changes when there are different opinions.

Reply:
> @tcharding perhaps you can save some time by waiting for resolution of our discussions before implementing changes when there are different opinions.

Yes, I'll try this and no be so trigger happy, cheers.

Reply:
@Kixunil what if we said "We do not implement `Ord` on this type because there is no natural ordering on opcodes, but there may appear to be one (e.g. because all the push opcodes appear in a consecutive block) and we don't want to encourage subtly buggy code. Please use the associated methods on `Opcode` to distinguish different types of opcodes."

If we wanted to mention Core, we could also point out that Bitcoin Core's `IsPushOnly` considers `OP_RESERVED` to be a "push opcode", allowing this opcodes in contexts where only pushes are supposed to be allowed, because of exactly this category of mistakes.

Reply:
Yes, I like that wording. The core example is a good one but not sure if we should add it to doc comment. Maybe hide it in `<details><summary>Example of Core bug caused by assuming ordering</summary> ... </details>`.